Check/Abstract.php

```diff
@@ -279,4 +279,67 @@ public function getPercentOverride() {
     return $this->percentOverride;
   }
 
+  /**
+   * Returns the values of the valid options for a command.
+   *
+   * This function is used by the codebase report where it needs to get the
+   * value of options for different tools it uses. The function takes an
+   * array of valid options to check and a prefix for those options.
+   *
+   * @param array $options
+   *   An array containing the options to be checked and their default values.
+   * @param string $option_prefix
+   *   Prefix for the options.
+   *
+   * @return array
+   *   An associative array containing the value of the options indexed by
+   *   option name.
+   */
+  public function getOptions(array $options, $option_prefix) {
+    $values = array();
+    foreach ($options as $option => $default) {
+      $value = drush_get_option($option_prefix . $option, $default);
+      if ($value !== NULL) {
+        $values[$option] = $value;
+      }
+    }
+    return $values;
+  }
+
+  /**
+   * Returns an array containing custom code paths or AUDIT_CHECK_SCORE_INFO.
```

Comment:
Instead of returning AUDIT_CHECK_SCORE_INFO, just make the function return the paths or an empty array.

Validate the paths to make sure that they are proper directories and they exist.

Use static so you only do these checks and validation once.

Then add a validation step to drush_site_audit_audit_codebase_validate for the custom code paths and return an error if the paths aren't correct.